public/components/common/Hint.tsx

```diff
@@ -10,41 +10,24 @@ interface HintProps {
   condition?: boolean;
 }
 
-interface HintState {
-  isClosed: boolean;
-}
-
-export class Hint extends React.Component<HintProps, HintState> {
-  private cacheKey: string | undefined;
-
-  constructor(props: HintProps) {
-    super(props);
-    this.cacheKey = this.props.permanentCloseKey ? `Hint-Closed-${this.props.permanentCloseKey}` : undefined;
-    this.state = {
-      isClosed: this.cacheKey ? cache.local.has(this.cacheKey) : false
-    };
-  }
-
-  private close = () => {
-    if (this.cacheKey) {
-      cache.local.set(this.cacheKey, "true");
+export const Hint:React.FC<HintProps> = props => {
+  let cacheKey: string | undefined = props.permanentCloseKey ? `Hint-Closed-${props.permanentCloseKey}` : undefined;
```

Comment:
cacheKey should probably be a const as well

Reply:
Yes I agree.